agent/engine/docker_image_manager.go

```diff
@@ -72,6 +74,18 @@ func NewImageManager(cfg *config.Config, client DockerClient, state dockerstate.
 	}
 }
 
+// PreserveImageUnsafe adds the image into preservedImages array that won't be cleaned up by agent
+func (imageManager *dockerImageManager) PreserveImageUnsafe(image string) {
```

Comment:
All of this additional unsafe code here makes me think that we should just do the pullImage change where we don't track add images of internal containers. Can we just do that and get rid of all of the changes in image manager instead? Otherwise, we are simply adding code knowing that we will remove it for sure in the future.